test/functional/feature_loadblock.py

```diff
@@ -0,0 +1,82 @@
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test loadblock option
+
+Test the option to start a node with the option loadblock which loads
+a serialized blockchain from a file (usually called bootstrap.dat).
+To generate that file this test uses the helper scripts available
+in contrib/linearize.
+"""
+
+import os
+import shutil
+import urllib
+
+from test_framework.test_framework import (
+    BitcoinTestFramework,
+    initialize_datadir,
+)
+from test_framework.util import assert_equal, wait_until
+
+
+class LoadblockTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def run_test(self):
+        self.nodes[0].generate(100)
+
+        # parsing the url of our node to get settings for config file
+        data_dir = self.nodes[0].datadir
+        node_url = urllib.parse.urlparse(self.nodes[0].url)
+        cfg_path = os.path.join(data_dir, "linearize.cfg")
+        hashlist_path = os.path.join(data_dir, "hashlist.txt")
+        bootstrap_file = os.path.join(self.options.tmpdir, "bootstrap.dat")
+        genesis_block = self.nodes[0].getblockhash(0)
+        block_dir = os.path.join(data_dir, "regtest", "blocks")
+
+        self.log.info("Create linearization config file")
+        with open(cfg_path, "a", encoding="utf-8") as cfg:
+            cfg.write("datadir={}\n".format(data_dir))
+            cfg.write("rpcuser={}\n".format(node_url.username))
+            cfg.write("rpcpassword={}\n".format(node_url.password))
+            cfg.write("port={}\n".format(node_url.port))
+            cfg.write("host={}\n".format(node_url.hostname))
+            cfg.write("output_file={}\n".format(bootstrap_file))
+            cfg.write("max_height=100\n")
+            cfg.write("netmagic=fabfb5da\n")
+            cfg.write("input={}\n".format(block_dir))
+            cfg.write("genesis={}\n".format(genesis_block))
+            cfg.write("hashlist={}\n".format(hashlist_path))
+
+        linearize_dir = os.path.join(self.config["environment"]["SRCDIR"], "contrib", "linearize")
+
+        self.log.info("Run linearization of block hashes")
+        linearize_hashes_path = os.path.join(linearize_dir, "linearize-hashes.py")
+        os.system(linearize_hashes_path + " " + cfg_path + " > " + hashlist_path)
+
+        self.log.info("Run linearization of block data")
+        linearize_data_path = os.path.join(linearize_dir, "linearize-data.py")
+        os.system(linearize_data_path + " " + cfg_path)
+
+        self.log.info("Restart node with bootstrap file")
+        self.stop_node(0)
+
+        # use a new empty blocks dir to make sure blocks are loaded from bootstrap file
+        shutil.rmtree(self.nodes[0].datadir)
+        initialize_datadir(self.options.tmpdir, 0, self.chain)
+        blocksdir_path = os.path.join(self.options.tmpdir, 'blocksdir')
+        os.mkdir(blocksdir_path)
```

Comment:
What is the point of `blocksdir_path` (new empty blocks dir), when the whole datadir is cleaned?

I'd even go further and not remove the whole datadir tree. Since that also deletes the debug log for debugging purposes.

You could spin up two nodes, and call `setnetworkactive(False)` on the second one. Then reboot the second one with the `-loadblock` arg here.

Reply:
I was weighing different options initially and thought using just one node might be more economical. But I did not think about the logs. I am now using a second node.